### PR TITLE
Switch from JSHint to ESLint

### LINKED ISSUE #7: Switch to ESLint
ESLint is objectively better (and more configurable).
Plus, it found these:

```
➜  fathom git:(master) ✗ npm run lint

> @ lint /Users/pdehaan/dev/github/fathom
> eslint .


/Users/pdehaan/dev/github/fathom/fathom.js
    6:7   error  'flatMap' is defined but never used       no-unused-vars
   10:7   error  'jsdom' is defined but never used         no-unused-vars
  222:44  error  'isBlock' is not defined                  no-undef
  243:10  error  'paragraphish' is defined but never used  no-unused-vars
  247:24  error  'str' is defined but never used           no-unused-vars

/Users/pdehaan/dev/github/fathom/test/test.js
  20:40  error  'node' is defined but never used  no-unused-vars
  35:30  error  'node' is defined but never used  no-unused-vars
  36:28  error  'node' is defined but never used  no-unused-vars

✖ 8 problems (8 errors, 0 warnings)
```

And my .eslintrc file looks like this:

```yaml
env:
  es6: true
  mocha: true
  node: true

extends:
  - eslint:recommended
```

### PR DESCRIPTION
**Pull request by pdehaan**
Fixes #7

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -1,16 +1,16 @@
...
-	@node_modules/.bin/jshint fathom.js
+	@npm run lint
 
 test:
-	@node_modules/.bin/mocha
+	@npm run test
 
 debugtest:
-	@node_modules/.bin/mocha -d
+	@npm run test -- -d
```
Comment (pdehaan) [inline Makefile]: Not sure if the Makefile adds anything here that we can't achieve with just the npm scripts tasks.
But I'm just a drive-by person walking in here and kicking up dust.

File: .eslintrc [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+  es6: true
+  mocha: true
+  node: true
+
+extends:
+  - eslint:recommended
+
+rules:
+  no-var: 0
+  prefer-const: 0
```
Comment (pdehaan) [inline .eslintrc]: Nit, we should also add `indent: [2, 4]` since there was at least 2-3 whitespace violations.

**Comment by pdehaan** [comment]
Anyways, if any of this is useful and agreeable to you, I can do a couple more [future] PRs for some ES6 optimizations, such as **destructuring**:

```js
--- a/test/test.js
+++ b/test/test.js
@@ -1,11 +1,7 @@
 const assert = require('chai').assert;
-const jsdom = require('../node_modules/jsdom');
+const jsdom = require('jsdom');

-const fathom = require('../fathom'),
-      dom = fathom.dom,
-      rule = fathom.rule,
-      ruleset = fathom.ruleset,
-      flavor = fathom.flavor;
+const { dom, rule, ruleset, flavor } = require('../fathom');
```

and **shorthand**: 

```js
 module.exports = {
-    dom: dom,
-    rule: rule,
-    ruleset: ruleset,
-    flavor: flavor
+    dom,
+    rule,
+    ruleset,
+    flavor
 };
```

File: package.json [hunk 3]
```diff
@@ -1,17 +1,30 @@
 {
-  "author": {
-    "name": "Erik Rose",
-    "url": "https://grinchcentral.com"
+  "name": "fathom",
```
Comment (pdehaan) [inline package.json]: I think `name` is a required field in package.json, but not sure if we want to go w/ "fathom", since it's already used in npm.
Or maybe we get fancy and give it a [scoped name](https://docs.npmjs.com/misc/scope), like `@mozilla/fathom` to avoid collisions.

**Comment by pdehaan** [comment]
Closing. Fixed via 9cf04ff4a09d90049e5f7bcffd35bb3bf4971ae4

File: Makefile (hunk 1 above)
Comment (erikrose) [inline Makefile]: Probably not at the moment. There might have been some in there at the beginning of the project, or it could just be me being new to the ecosystem; I wasn't aware of npm scripts.

**Comment by erikrose** [comment]
There was a lot of good stuff in here that didn't get merged! I shall harvest it. :-)

**Comment by erikrose** [comment]
Harvested!

File: package.json (hunk 3 above)
Comment (erikrose) [inline package.json]: We'll have to tweak the name. I lean away from a scoped name—at least one with an org name in it—because history has shown that projects have a tendency to migrate from org to org. Ask anybody still prepending "NS" to Cocoa APIs. :-)
